code/modules/unit_tests/fish_unit_tests.dm

```diff
@@ -123,5 +123,57 @@
 	. = ..()
 	probability = 0 //works around the global list initialization skipping abstract/impossible evolutions.
 
+/// Checks that we are able to fish people out of chasms with priority and that they end up in the right location
+/datum/unit_test/fish_rescue_hook
+
+/datum/unit_test/fish_rescue_hook/Run()
+	var/mob/living/carbon/human/consistent/get_in_the_hole = allocate(/mob/living/carbon/human/consistent) // create a human dummy to drop in the chasm
+	var/mob/living/basic/mining/lobstrosity/you_too = allocate(/mob/living/basic/mining/lobstrosity) // create some mindless mobs to fill the contents a bit
+	var/mob/living/carbon/human/consistent/mindless = allocate(/mob/living/carbon/human/consistent)
+	var/mob/living/carbon/human/consistent/no_brain = allocate(/mob/living/carbon/human/consistent)
+	var/mob/living/carbon/human/consistent/empty = allocate(/mob/living/carbon/human/consistent)
+	var/mob/living/carbon/human/consistent/dummy = allocate(/mob/living/carbon/human/consistent)
+
+	// our 'fisherman' where we expect the item to be moved to after fishing it up
+	var/mob/living/carbon/human/consistent/a_fisherman = allocate(/mob/living/carbon/human/consistent, run_loc_floor_top_right)
+
+	// create our chasm and remember the previous turf so we can change it back once we're done
+	var/turf/initial_turf_type = run_loc_floor_bottom_left.type
+	var/turf/open/chasm/the_hole = allocate(/turf/open/chasm)
+
+	// into the hole they go
+	the_hole.drop(get_in_the_hole)
+	the_hole.drop(you_too)
+	the_hole.drop(mindless)
+	the_hole.drop(no_brain)
+	the_hole.drop(empty)
+	the_hole.drop(dummy)
+
+	// pretend like this mob has a mind. they should be fished up first
+	get_in_the_hole.mind = TRUE
+
+	SEND_SIGNAL(the_hole, COMSIG_PRE_FISHING) // we need to do this for the fishing spot component to be attached
+	var/datum/component/fishing_spot/the_hole_fishing_spot = the_hole.GetComponent(/datum/component/fishing_spot)
+	var/datum/fish_source/fishing_source = the_hole_fishing_spot.fish_source
```

Comment:
To be frank, you could simply use any preset fishing spot like this:
```
	var/datum/fish_source/fishing_source = GLOB.preset_fish_sources[/datum/fish_source/chasm]
```
but I understand why you would go through the long way.